### PR TITLE
chore(cli-repl): adds deprecation warnings on startup for mongosh 2.x.x when running on unsupported platforms - MONGOSH-1553

### PR DESCRIPTION
**Pull request by himanshusinghs**
A screenshot of how it will look:

image

### CODE REVIEW

File: packages/cli-repl/src/cli-repl.ts [hunk 1]
```diff
@@ -795,6 +798,75 @@ export class CliRepl implements MongoshIOProvider {
...
+      warnings.push(
+        '  - Using mongosh with Node.js versions lower than 20.0.0 is deprecated, and support will be removed in a future release.'
+      );
+    }
+
+    if (warnings.length) {
+      const deprecationWarning = [
+        'Deprecation warnings:',
+        ...warnings,
+        'See https://www.mongodb.com/docs/mongodb-shell/ for documentation on supported platforms.',
```
Comment (himanshusinghs) [inline packages/cli-repl/src/cli-repl.ts:861]: This I will confirm with @GaurabAryal wether it is ok to keep this or if already has a link for me to put here.

Comment (himanshusinghs) [reply packages/cli-repl/src/cli-repl.ts:862]: Confirmed that we're leaving the link as is for now but I will create a follow up ticket for this link to be updated before releasing 2.x

**Comment by himanshusinghs** [comment]
Evergreen Patch - https://spruce.mongodb.com/version/64db280b2a60edbc37b07993/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: packages/cli-repl/src/cli-repl.ts [hunk 2]
```diff
@@ -795,6 +798,76 @@ export class CliRepl implements MongoshIOProvider {
...
+        }
+      | undefined;
+    if (!processReport) {
+      return;
+    }
+
+    const warnings: string[] = [];
+    const RECOMMENDED_GLIBC = '>=2.28.0';
+    const RECOMMENDED_OPENSSL = '>=3.0.0';
+    const RECOMMENDED_NODEJS = '>=20.0.0';
```
Comment (himanshusinghs) [inline packages/cli-repl/src/cli-repl.ts:822]: I was not sure if there is a better place to put these constants in so I kept them here.

File: packages/cli-repl/src/cli-repl.ts [hunk 3]
```diff
@@ -795,6 +798,76 @@ export class CliRepl implements MongoshIOProvider {
...
+    if (this.cliOptions.quiet) {
+      return;
+    }
+
+    // Typings for process.getReport haven't been updated
+    // (https://github.com/DefinitelyTyped/DefinitelyTyped/issues/40140)
+    const processReport = process.report?.getReport() as unknown as
+      | {
+          header: {
+            glibcVersionRuntime: string;
```
Comment (addaleax) [inline packages/cli-repl/src/cli-repl.ts:811]:
```suggestion
            glibcVersionRuntime?: string;
```

(not present on non-Linux platforms)

**Review by addaleax — APPROVED** [review]

**Comment by himanshusinghs** [comment]
Evergreen Patch - https://spruce.mongodb.com/version/64db494f0ae60688ec81ba6c/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Comment by himanshusinghs** [comment]
Latest Patch - https://spruce.mongodb.com/version/64db94df0ae60688ec824d01/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Pull request merged**